### PR TITLE
platform: Support reading yaml file from file system, classpath, and url

### PR DESCRIPTION
**Pull request by lijamie98**
<details>
  <summary>PR Checklist</summary>

### PR Structure

* [x] This PR has reasonably narrow scope (if not, break it down into smaller PRs).
* [x] This PR avoids mixing refactoring changes with feature changes (split into two PRs
  otherwise).
* [x] This PR's title starts with name of package that is most changed in the PR, ex.
  `paymentservice.stellar`, or `all` or `doc` if the changes are broad or impact many
  packages.

### Thoroughness

* [x] This PR adds tests for the most critical parts of the new functionality or fixes.
</details>

### What

The yaml file can be read from one of the three sources:
* file system
* classpath
* url, such as http, https

### Why

It is possible that the yaml file exist outside the local file system.
When doing unit testing, the test yaml file is in `resources` folder which can be read from the classpath.

### CODE REVIEW

**Review by JakeUrban — APPROVED** [review]
This is great. One thing I think is super important is to have documentation regarding the format and accepted values for the config file. Not a blocker here but definitely a blocker for release.

**Comment by lijamie98** [comment]
The definition is still evolving. We should do a documentation review when we are close to release.

@reecexlm Can we put this in the project plan?

**Comment by lijamie98** [comment]
@marcelosalloum Can you focus checking the `payment-gateway` section? Currently it does not support two services of the same kind. For example, we can not have two services with "CirclePaymentService" or "StellarPaymentService".

File: platform/src/test/resources/test-anchor-config.yaml [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+      stellarNetwork: TESTNET
+
+    #
+    # Payment Stellar configurations
+    #
+    stellar:
+      enabled: false
+      name: "stellar"
+      horizonUrl: https://horizon-testnet.stellar.org
+      secretKey: secret
```
Comment (marcelosalloum) [inline platform/src/test/resources/test-anchor-config.yaml:69]: @lijamie98, about your comment:
https://github.com/stellar/java-stellar-anchor-sdk/pull/63#issuecomment-1043344235

I think this is fine for now. Let's not overthink this and move forward... making the stellar property accept an array is something we can do later, when we really need it.

File: build.gradle.kts [hunk 2]
```diff
@@ -8,7 +8,7 @@ subprojects {
   apply(plugin = "com.diffplug.spotless")
 
   group = "org.stellar.anchor-sdk"
-  version = "0.1.0"
+  version = "1.0.2"
```
[the remark this answers is not in the document]
Comment (lijamie98) [reply build.gradle.kts:11]: Yes, that's right. The problem is when I publish to Maven Central, I used version 1.0.1. And Maven Central does not allow me to remove the mistakenly published artifact for stability of the community. 
https://issues.sonatype.org/browse/OSSRH-78131

I will set the version back to 0.1.1 and detach the core published version with the sdk version.

File: platform/src/main/java/org/stellar/anchor/server/ConfigConfiguration.java [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+import org.stellar.anchor.paymentservice.circle.config.StellarPaymentConfig;
+import org.stellar.anchor.server.config.PropertyAppConfig;
+import org.stellar.anchor.server.config.PropertySep10Config;
+import org.stellar.anchor.server.config.PropertySep1Config;
+import org.stellar.anchor.server.config.PropertySep24Config;
+import org.stellar.anchor.server.config.payment.PropertyCirclePaymentConfig;
+import org.stellar.anchor.server.config.payment.PropertyStellarPaymentConfig;
+
+@Configuration
+public class ConfigConfiguration {
```
[the remark this answers is not in the document]
Comment (lijamie98) [reply platform/src/main/java/org/stellar/anchor/server/ConfigConfiguration.java:20]: This is a result of a name overloading. Spring use @Configuration for Spring configuration. What we are doing here is for the Configuration Management. Therefore, we have ConfigConfiguration.

This class is the aggregation of all the application configuration management beans. These names `AppConfiguration, GlobalConfiguration, AnchorConfiguration or PlatformConfiguration` will not fit.

I will use `ConfigManagementConfiguration` for now and I am open to future suggestions.

[the remark this answers is not in the document]
Comment (marcelosalloum) [reply platform/src/main/java/org/stellar/anchor/server/ConfigConfiguration.java]: `ConfigManagementConfiguration` sounds clearer, thanks for explaining.

**Pull request merged**